Bottom: f16af6c0376a344bff88f7df5175a2e9d9718c68
Top:    ba49cee347e2eef25fac974fef1de545a870e529
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-03 02:13:47 +0000

Refresh of improve-gl-stuff

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 5d4ae3f..6bcc47a 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -60,28 +60,10 @@ hidgl_new_triangle_array (void)
 void
 hidgl_init_triangle_array (triangle_buffer *buffer)
 {
-  GLenum errCode;
-  const GLubyte *errString;
-
+  glEnableClientState (GL_VERTEX_ARRAY);
+  glVertexPointer (2, GL_FLOAT, 0, buffer->triangle_array);
   buffer->triangle_count = 0;
   buffer->coord_comp_count = 0;
-
-  glEnableClientState (GL_VERTEX_ARRAY);
-  glGenBuffers (1, &buffer->vbo_name);
-  glBindBuffer (GL_ARRAY_BUFFER, buffer->vbo_name);
-  glBufferData (GL_ARRAY_BUFFER, TRIANGLE_ARRAY_BYTES, NULL, GL_STATIC_DRAW);
-
-  buffer->triangle_array = glMapBuffer (GL_ARRAY_BUFFER, GL_WRITE_ONLY);
-
-  if ((errCode = glGetError()) != GL_NO_ERROR) {
-      errString = gluErrorString(errCode);
-     fprintf (stderr, "OpenGL Error: %s\n", errString);
-  }
-
-  if (buffer->triangle_array == NULL) {
-    printf ("Couldn't map VBO.. sorry, don't know how best to handle this gracefully\n");
-    exit (1);
-  }
 }
 
 void
@@ -90,18 +72,9 @@ hidgl_flush_triangles (triangle_buffer *buffer)
   if (buffer->triangle_count == 0)
     return;
 
-  glUnmapBuffer (GL_ARRAY_BUFFER);
-
-  glVertexPointer (2, GL_FLOAT, 0, NULL); // buffer->triangle_array);
   glDrawArrays (GL_TRIANGLES, 0, buffer->triangle_count * 3);
-
-//  buffer->triangle_count = 0;
-//  buffer->coord_comp_count = 0;
-
-  glBindBuffer(GL_ARRAY_BUFFER, 0);
-  glDeleteBuffers (1, &buffer->vbo_name);
-
-  hidgl_init_triangle_array (buffer);
+  buffer->triangle_count = 0;
+  buffer->coord_comp_count = 0;
 }
 
 void
diff --git a/src/hid/common/hidgl.h b/src/hid/common/hidgl.h
index f82056a..1f15d3f 100644
--- a/src/hid/common/hidgl.h
+++ b/src/hid/common/hidgl.h
@@ -31,11 +31,9 @@
    4 * 5461 * 2 * 3 = 109464 */
 #define TRIANGLE_ARRAY_BYTES 131072
 typedef struct {
-//  GLfloat triangle_array [2 * 3 * TRIANGLE_ARRAY_SIZE];
-  GLfloat *triangle_array;
+  GLfloat triangle_array [2 * 3 * TRIANGLE_ARRAY_SIZE];
   unsigned int triangle_count;
   unsigned int coord_comp_count;
-  GLuint vbo_name;
 } triangle_buffer;
 
 extern triangle_buffer buffer;
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 17ecae2..7738098 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1077,11 +1077,6 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
   hidgl_flush_triangles (&buffer);
   glPopMatrix ();
 
-
-  glUnmapBuffer (GL_ARRAY_BUFFER);
-  glBindBuffer(GL_ARRAY_BUFFER, 0);
-  glDeleteBuffers (1, &buffer.vbo_name);
-
   draw_grid ();
 
   hidgl_init_triangle_array (&buffer);
@@ -1102,10 +1097,6 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
 
   hidgl_flush_triangles (&buffer);
 
-  glUnmapBuffer (GL_ARRAY_BUFFER);
-  glBindBuffer(GL_ARRAY_BUFFER, 0);
-  glDeleteBuffers (1, &buffer.vbo_name);
-
   if (gdk_gl_drawable_is_double_buffered (pGlDrawable))
     gdk_gl_drawable_swap_buffers (pGlDrawable);
   else
diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index bebaf96..137471f 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -236,10 +236,6 @@ ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * ev)
   hidgl_flush_triangles (&buffer);
   glPopMatrix ();
 
-  glUnmapBuffer (GL_ARRAY_BUFFER);
-  glBindBuffer(GL_ARRAY_BUFFER, 0);
-  glDeleteBuffers (1, &buffer.vbo_name);
-
   if (gdk_gl_drawable_is_double_buffered (pGlDrawable))
     gdk_gl_drawable_swap_buffers (pGlDrawable);
   else
